### ISSUE TITLE
Two regions at same admin_level with same name

### ISSUE THREAD

**Issue by morandd** [AndGem/osm_extract_polygon#10]
The "name" property of the admin_level relation is, unfortunately, not always unique. In Spain for example we find that at admin_level 8 the following two towns have the same 'name' property. (Parenthetically, I note some other data quality issues in the Relations, like the property "population" is defined twice, once as a number and then again as an object with subproperties.)

1. Google Maps Name: 28939 Arroyomolinos, Madrid, Spain
OSM name: "Arroyomolinos"
OSM Relation ID: "relation/341636"
2. Google Maps Name: Arroyomolinos, Cáceres, Cáceres, Spain
OSM name: "Arroyomolinos"
OSM Relation ID: "relation/1864624"

So when I run this tool for Spain at admin_level=8, the first Arroyomolinos.poly gets overwritten by the second and is thus lost.

A solution could be to suffix either "_1", "_2", etc, or the @id property (like "relation/1864624") to nonunique output .poly filenames. Personally I'd opt for the former suffixing style.

Thanks for this great tool!

**Comment by AndGem** [comment]
Hi @morandd ,

as I have said already in your PR, I'm really grateful for your very valuable Feedback and a solution approach. It really helped to have an example PR (even if not 100% correct) to think about how to implement a solution to the problem. While I got inspired by your PR, I chose a slightly different approach because with it I was able to fix another issue that I had open for way too long. But really thanks for this.

I've just updated the project and bumped its version to `0.2.0`. My tests with the spanish osm file (thanks for the pointer) already seemed to confirm that this case is now handled properly, but I would like to get the feedback from you to see if you can confirm everything is fine now.

Once you can confirm the issue is fixed, we can close this issue.

Thanks again!

**Comment by morandd** [comment]
Hi,
Thank you so much for the fix! I am testing it now (long runtime) but suppose that it works.
There are two small niggles with the fix: one is that the capitalization of region names is lost, and they become all lower case. The test for duplicate names should be case insensitive but it would be nice to preserve the capitalization, since human readers of the output do expect to see normal capitalization and there's no software way for downstream routines to recover the capitalization info.

The other is the Conflict mode - you introduced a user prompt, but could we also have a command line option? I guess I am not the only person who'd like to use this in batch mode. (My use is to extract all regions at all levels across Europe, so batch mode is quite nice/vital).

Thanks,
Dan

**Comment by AndGem** [comment]
Hi Dan,

thanks for the feedback! I agree to both points. 

The second point I had anyway on my to do list. The idea behind having the conflict mode was to ensure that we do not overwrite anything that was create outside of the program (maybe by a different run of the program or something else). Just to be more defensive there, but I also thought about the typical use-case and that probably you want to control this from the parameters directly.

The first point I actually only added passing by, just because I saw that this fixed an issue of overwriting files (similar to what you reported). Frankly, I did not expect anyone to notice, but fair enough. I'll fix it as well.

I'll create two issues and patch the program in the coming days.
